pinot-controller/pom.xml

```diff
@@ -33,7 +33,7 @@
   <url>https://pinot.apache.org/</url>
   <properties>
     <pinot.root>${basedir}/..</pinot.root>
-    <npm.script>build</npm.script>
+    <npm.script>build-ci</npm.script>
```

Comment:
Can we undo this change. This has effectively made it so everyone building this project outside of that one Dockerfile is actually building dev. I randomly caught this on one of our clusters because we (and probably others) build from source ourselves.

Reply:
TBH that was the intention. The idea is that unless you want to build a production bundle, you should build the dev version.

I don't think we need to rollback this, but what we can do is to run `npm build` when profile `bin-dist` is enabled, as that is the _maven way_ to say that we want to have a production bundle.

Reply:
I've changed this in #10525